package/piksi_system_daemon/src/main.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Gareth McMullin <gareth@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <libsbp/sbp.h>
+#include <libsbp/piksi.h>
+#include <libsbp/logging.h>
+#include <czmq.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "settings.h"
+#include "sbp_zmq.h"
+
+static int uart0_baudrate = 115200;
+static int uart1_baudrate = 115200;
+
+bool baudrate_notify(struct setting *s, const char *val)
+{
+  int baudrate;
+  bool ret = s->type->from_string(s->type->priv, &baudrate, s->len, val);
+  if (!ret) {
+    return false;
+  }
+
+  const char *dev = NULL;
+  if (s->addr == &uart0_baudrate) {
+    dev = "/dev/ttyPS0";
+  } else if (s->addr == &uart1_baudrate) {
+    dev = "/dev/ttyPS1";
+  } else {
+    return false;
+  }
+  char cmd[80];
+  snprintf(cmd, sizeof(cmd), "stty -F %s %d", dev, baudrate);
+  ret = system(cmd) == 0;
+
+  if (ret) {
+    *(int*)s->addr = baudrate;
+  }
+  return ret;
+}
+
+static const char const * ip_mode_enum[] = {"Static", "DHCP", NULL};
+static struct setting_type ip_mode_settings_type;
+static int TYPE_IP_MODE = 0;
+enum {IP_CFG_STATIC, IP_CFG_DHCP};
+u8 eth_ip_mode = IP_CFG_STATIC;
+static char eth_ip_addr[16] = "192.168.0.222";
+static char eth_netmask[16] = "255.255.255.0";
+static char eth_gateway[16] = "192.168.0.1";
+
+static void eth_update_config(void)
+{
+  system("ifdown eth0");
+
+  FILE *interfaces = fopen("/etc/network/interfaces", "w");
```

Comment:
Oh, yes it does. I guess as long as all the network interface settings (including possibly USB ethernet, for example) are contained in this daemon, we can just overwrite the whole file every time. Sounds fine.